src/halina/email_rapport_service.py

```diff
@@ -46,8 +74,41 @@ async def _collect_data_and_send(self):
         logger.info(f"Scanning stream for fits completed. "
                     f"Find fits: {[f"{name}: ({i.count_fits})" for name, i in telescopes.items()]}")
 
-        # todo continue here
+        # Prepare data for email
+        telescope_data = []
+        all_data_objects = defaultdict(lambda: DataObject(name="", count=0, filters=set()))
 
+        for tel in self._telescopes:
+            telescope_info = {
+                'name': tel,
+                'count_fits': telescopes[tel].count_fits,
+                'count_fits_processed': telescopes[tel].count_fits_processed,
+                'malformed_raw_count': telescopes[tel].malformed_raw_count,
+                'malformed_zdf_count': telescopes[tel].malformed_zdf_count
+            }
+            telescope_data.append(telescope_info)
+            merged_objects = self.merge_data_objects(telescopes[tel].objects)
+            for key, value in merged_objects.items():
+                all_data_objects[key].name = value.name
+                all_data_objects[key].count += value.count
+                all_data_objects[key].filters.update(value.filters)
 
-class SendEmailException(Exception):
-    pass
+        # Build and send email
+        night = self.format_night()
+
+        email_builder = EmailBuilder()
```

Comment:
nie o to chodzi w metodach buildera, w ten sposób używa się seterów linia po lini i za każdym razem odwołując się od nowa do email_builder. Te specyficzne metody wywołujesz tak:
email_builder.subject("subject").night(night).costam("costam")
można je zrobić w jednej linijce albo dla czytelności karzde ustawienie w osobnej ale bez wskazywania az każdym razem email_builder